processor/src/main/java/com/linecorp/decaton/processor/processors/BatchingProcessor.java

```diff
@@ -56,14 +57,14 @@ public static class BatchingTask<T> {
 
     /**
      * Instantiate {@link BatchingProcessor}.
-     * @param lingerMillis time limit for this processor. On every lingerMillis milliseconds,
+     * @param lingerMillisSupplier time limit for this processor. On every lingerMillis milliseconds,
      * tasks in past lingerMillis milliseconds are pushed to {@link BatchingTask#processBatchingTasks(List)}.
-     * @param capacity size limit for this processor. Every time tasks’size reaches capacity,
+     * @param capacitySupplier size limit for this processor. Every time tasks’size reaches capacity,
      * tasks in past before reaching capacity are pushed to {@link BatchingTask#processBatchingTasks(List)}.
      */
-    protected BatchingProcessor(long lingerMillis, int capacity) {
-        this.lingerMillis = lingerMillis;
-        this.capacity = capacity;
+    protected BatchingProcessor(Supplier<Long> lingerMillisSupplier, Supplier<Integer> capacitySupplier) {
```

Comment:
I suggest keep current constructor to avoid breaking change, by adding overload like `BatchingProcessor(lingerMillis, capacity) { this(() -> lingerMillis, () -> capacity) ... }`